protocol/messenger_profile_showcase.go

```diff
@@ -104,26 +132,76 @@ func fromProfileShowcaseCollectibleProto(messages []*protobuf.ProfileShowcaseCol
 	collectibles := []*ProfileShowcaseCollectible{}
 	for _, entry := range messages {
 		collectibles = append(collectibles, &ProfileShowcaseCollectible{
-			UID:   entry.Uid,
-			Order: int(entry.Order),
+			ContractAddress: entry.ContractAddress,
+			ChainID:         entry.ChainId,
+			TokenID:         entry.TokenId,
+			CommunityID:     entry.CommunityId,
+			AccountAddress:  entry.AccountAddress,
+			Order:           int(entry.Order),
 		})
 	}
 	return collectibles
 }
 
-func fromProfileShowcaseAssetProto(messages []*protobuf.ProfileShowcaseAsset) []*ProfileShowcaseAsset {
-	assets := []*ProfileShowcaseAsset{}
+func fromProfileShowcaseVerifiedTokenProto(messages []*protobuf.ProfileShowcaseVerifiedToken) []*ProfileShowcaseVerifiedToken {
+	tokens := []*ProfileShowcaseVerifiedToken{}
 	for _, entry := range messages {
-		assets = append(assets, &ProfileShowcaseAsset{
+		tokens = append(tokens, &ProfileShowcaseVerifiedToken{
 			Symbol: entry.Symbol,
 			Order:  int(entry.Order),
 		})
 	}
-	return assets
+	return tokens
+}
+
+func fromProfileShowcaseUnverifiedTokenProto(messages []*protobuf.ProfileShowcaseUnverifiedToken) []*ProfileShowcaseUnverifiedToken {
+	tokens := []*ProfileShowcaseUnverifiedToken{}
+	for _, entry := range messages {
+		tokens = append(tokens, &ProfileShowcaseUnverifiedToken{
+			ContractAddress: entry.ContractAddress,
+			ChainID:         entry.ChainId,
+			Order:           int(entry.Order),
+		})
+	}
+	return tokens
+}
+
+func Validate(preferences *ProfileShowcasePreferences) error {
+	if (len(preferences.VerifiedTokens) > 0 || len(preferences.UnverifiedTokens) > 0 || len(preferences.Collectibles) > 0) &&
+		len(preferences.Accounts) == 0 {
+		return errorNoAccountProvidedWithTokenOrCollectible
+	}
+
+	accountsMap := make(map[string]*ProfileShowcaseAccountPreference)
+	for _, account := range preferences.Accounts {
+		if _, ok := accountsMap[account.Address]; ok {
+			return errorDublicateAccountAddress
+		}
+		accountsMap[account.Address] = account
+	}
+
+	for _, collectible := range preferences.Collectibles {
+		if account, ok := accountsMap[collectible.AccountAddress]; ok {
+			if account.ShowcaseVisibility < collectible.ShowcaseVisibility {
+				return errorAccountVisibilityLowerThanCollectible
+			}
+		} else {
+			return nil
+			// NOTE: with current wallet collectible implementation we don't know account on this stage
+			// return errorNoAccountAddressForCollectible
+		}
```

Comment:
Nothing important, just more readable IMHO:
```suggestion
        account, ok := accountsMap[collectible.AccountAddress]
        if !ok {
            return nil
			// NOTE: with current wallet collectible implementation we don't know account on this stage
			// return errorNoAccountAddressForCollectible
	    }
	    if account.ShowcaseVisibility < collectible.ShowcaseVisibility {
		    return errorAccountVisibilityLowerThanCollectible
	    }
```